### PR TITLE
[OHCI] Allow more than 16 devices

### PR DESCRIPTION
**Pull request by Ryzee119**
**Describe the PR**
Fixes issue we discussed way back in https://github.com/hathach/tinyusb/issues/1497
```
[Ryzee119]commented [on Jun 8, 2022]
I will create a PR for this fix soon
```
heh :upside_down_face: 

**Additional context**
This now allows 256 EDs, and >=127 devices as per USB spec. This should cover pretty much anyones needs while keeping a few bits in the TD structs for future use.

The extra array will likely help with iso transfers which I am look at now as we likely need to track more data

### CODE REVIEW

**Review by hathach — APPROVED** [review]
superb! Thank you very much for improving the ohci driver. There are some minor things, that I have noticed and take the chance to add cmake support to lpc17 to test out the PR and pushed some update to
- move gtd extra out of control pool to save sram
- also rename gtd_data to gtd_extra

Please let me know if these changes make sense to you.

Note: I haven't tested on actual hardware yet, if you have time please do on your side. We can merge this once tested.

File: src/portable/ohci/ohci.h [hunk 1]
```diff
@@ -164,11 +172,13 @@ typedef struct TU_ATTR_ALIGNED(256)
   struct {
     ohci_ed_t ed;
     ohci_gtd_t gtd;
+    gtd_extra_data_t gtd_data;
```
[the remark this answers is not in the document]
Comment (Ryzee119) [reply src/portable/ohci/ohci.h]: Youre right this a good pick up. Thanks

**Comment by Ryzee119** [comment]
Looks like you beat me to the changes. Good pick ups
I have tested on hardware is all is working great. I tested with 17 devices + 3 hubs :fire: 

```
HID MOUNTED 01 0
START HID
A device with address 1 is mounted
HID MOUNTED 02 1
START HID
A device with address 2 is mounted
HID MOUNTED 03 2
START HID
A device with address 3 is mounted
HID MOUNTED 04 3
START HID
A device with address 4 is mounted
A device with address 5 is mounted
A device with address 6 is mounted
A device with address 7 is mounted
A device with address 8 is mounted
A device with address 9 is mounted
A device with address 10 is mounted
A device with address 11 is mounted
A device with address 12 is mounted
A device with address 13 is mounted
A device with address 14 is mounted
A device with address 15 is mounted
A device with address 16 is mounted
A device with address 17 is mounted

A device with address 1 is unmounted 
HID UNMOUNTED 01 0
A device with address 2 is unmounted 
HID UNMOUNTED 02 1
A device with address 3 is unmounted 
HID UNMOUNTED 03 2
A device with address 4 is unmounted 
HID UNMOUNTED 04 3
A device with address 16 is unmounted 
A device with address 17 is unmounted 
A device with address 5 is unmounted 
A device with address 6 is unmounted 
A device with address 7 is unmounted 
A device with address 8 is unmounted 
A device with address 9 is unmounted 
A device with address 10 is unmounted 
A device with address 11 is unmounted 
A device with address 12 is unmounted 
A device with address 13 is unmounted 
A device with address 14 is unmounted 
A device with address 15 is unmounted 
```

**Comment by hathach** [comment]
superb! thank you for your Pr and testing it out. this is good to merge now.

**Pull request merged**